unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1691,7 +1691,12 @@ public Quaternion TargetRotation
 
             try {
                 ActionDispatcher.Dispatch(target: target, dynamicServerAction: controlCommand);
+            } catch (InvalidArgumentsException e) {
+                errorMessage = $"action: ${controlCommand.action} called with invalid arguments:" + string.Join(",", e.ArgumentNames.ToArray());
+                errorCode = ServerActionErrorCode.InvalidArgument;
+                actionFinished(false);
```

Comment:
Can we enrich this error message a bit by adding a list of all the ways the given action can be called? I.e. I would like to see an error message that returns something like:
```
You have attempted to use the "PushObject" action with invalid extraneous arguments.
Expected arguments: ...
Your arguments: ...
Invalid arguments: ...

Valid ways to invoke the for "PushObject" action include:
# 1 - PushObject(objectId, ...)
# 2 - PushObject(objectId, ...)
# 3 - PushObject(objectId, ...)
```

Reply:
that is a good idea let me see how much restructuring it needs to get all the overwrites, since at that point in the code we had selected a particular, best matching, overwrite